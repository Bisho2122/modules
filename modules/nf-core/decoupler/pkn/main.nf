process DECOUPLER_PKN {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "ghcr.io/saezlab/publish-packages/decoupler:sha-2f65a0d"

    input:
    tuple val(meta), val(resource), val(organism)

    output:
    tuple val(meta), path("*.csv") , emit: network
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix   = task.ext.prefix ?: "${meta.id}_decoupler_network"
    """
    #!/usr/bin/env python3

    import decoupler as dc
    import pandas as pd

    resource_dict = {'dorothea' : dc.get_dorothea,
                    'collectri' : dc.get_collectri,
                    'progeny'   : dc.get_progeny}

    resource_type = "${resource}".lower()
    org = "${organism}".lower()

    get_net_df = resource_dict.get(resource_type)

    net = get_net_df(organism = org)
    net.to_csv("${prefix}.csv")

    ## VERSIONS FILE
    with open('versions.yml', 'a') as version_file:
        version_file.write('"${task.process}":' + "\\n")
        version_file.write("\tdecoupler-py: " + dc.__version__ + "\\n")
    """
    stub:
    prefix   = task.ext.prefix ?: "${meta.id}_decoupler_network"
    """
    #!/usr/bin/env python3
    open("${prefix}.csv", 'a').close()

    with open('versions.yml', 'a') as version_file:
        version_file.write('"${task.process}":' + "\\n")
        version_file.write("\tdecoupler-py: " + dc.__version__ + "\\n")
    """
}
