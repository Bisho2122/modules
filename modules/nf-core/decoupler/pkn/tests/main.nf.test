nextflow_process {

    name "Test Process DECOUPLER_PKN"
    script "../main.nf"
    process "DECOUPLER_PKN"

    tag "modules"
    tag "modules_nfcore"
    tag "decoupler"
    tag "decoupler/pkn"

    test("decouplerPKN_test") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    'progeny',
                    'human'
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("decouplerPKN_test - stub") {
        options("--stub")

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    'progeny',
                    'human'
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
