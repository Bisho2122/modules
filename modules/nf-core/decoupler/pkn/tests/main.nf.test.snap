{
    "decouplerPKN_test - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_decoupler_network.csv:md5,9ca4fa1621927908276b964a966efc23"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3b582d61b8b2f5c7747be1bf1a7ef902"
                ],
                "network": [
                    [
                        {
                            "id": "test"
                        },
                        "test_decoupler_network.csv:md5,9ca4fa1621927908276b964a966efc23"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3b582d61b8b2f5c7747be1bf1a7ef902"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T16:08:01.794760683"
    },
    "decouplerPKN_test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_decoupler_network.csv:md5,9ca4fa1621927908276b964a966efc23"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3b582d61b8b2f5c7747be1bf1a7ef902"
                ],
                "network": [
                    [
                        {
                            "id": "test"
                        },
                        "test_decoupler_network.csv:md5,9ca4fa1621927908276b964a966efc23"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3b582d61b8b2f5c7747be1bf1a7ef902"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T16:05:15.156866696"
    }
}